### PR TITLE
fix(wasm): key tracking when no one is subscribing to Key<Down|Up>

### LINKED ISSUE #18617: [WASM] CalendarView of CalendarDatePicker does not work after entering uppercase letters in PasswordBox.
### Current behavior

After entering capital letters or special characters in PasswordBox, CalendarView of CalendarDatePicker does not work.

After entering capital letters, the calendar is no longer displayed unless you refresh.

https://github.com/user-attachments/assets/de5fda0b-8315-478d-8b09-ba28af849798

### Expected behavior

A calendar should appear when you enter something in the password box.

### How to reproduce it (as minimally and precisely as possible)

[UnoApp3.zip](https://github.com/user-attachments/files/17567435/UnoApp3.zip)

### Workaround

_No response_

### Works on UWP/WinUI

Yes

### Environment

Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### NuGet package version(s)

Uno.Sdk 5.4.10
Uno.WinUI 5.4.38
Uno.Bootstrap 8.0.21
Uno.Extensions 5.0.2
Uno.Toolkit 6.2.6
Uno.Resizetizer 1.6.0

### Affected platforms

WebAssembly

### IDE

Visual Studio 2022

### IDE version

17.11.5

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by ramezgerges**
GitHub Issue (If applicable): closes #18617

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by ramezgerges** [comment]
This affects NavigationView (as reported in the original issue) because, for some reason, NavigationView's logic is written so that if any modifiers are pressed, it should _not_ open the picker popup.

File: src/Uno.UWP/UI/Core/Internal/KeyboardStateTracker.cs [hunk 1]
```diff
@@ -106,4 +107,21 @@ internal static void Reset()
 			_keyStates.Clear();
 		}
 	}
+
+#if __WASM__
+#pragma warning disable IDE0051 // Remove unused private members
+	[JSExport]
+	private static void UpdateKeyStateNative(string key, bool down)
```
Comment (jeromelaban) [inline src/Uno.UWP/UI/Core/Internal/KeyboardStateTracker.cs:114]: Can you add a manual test that explains what to test? Thanks!

**Review by MartinZikmund — APPROVED** [review]
We shall likely need a similar solution on other targets as well, as currently the tracking is not completely reliable, e.g. when key is pressed before the window gets focused, opened https://github.com/unoplatform/uno/issues/18714 to track this.

**Comment by ramezgerges** [comment]
@MartinZikmund Actually, this is still the case on WASM after this PR. This only fixes missing tracking updates when KeyDown/Up has no listeners, but an out-of-focus press will still mess up the tracking. We will need to investigate that separately.

**Pull request merged**